File: Data Structures & Algorithms/two-integer-sum-ii/submission-4.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int[] ans = new int[2];
        int e = numbers.length - 1;

        for(int i=0; i<e; i++){
            if(numbers[i]+numbers[e] > target){
                i--;
                e--;
            }else if(numbers[i]+numbers[e] == target){
                ans[0] = i+1;
                ans[1] = e+1;
                break;
            }
        }

        return ans;
    }
}

File: Data Structures & Algorithms/search-2d-matrix/submission-4.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        if target > matrix[-1][-1] or target < matrix[0][0]:
            return False
        
        l = 0
        r = len(matrix[0]) * len(matrix) - 1

        while l <= r:
            m = (r+l) // 2 # 5

            val = matrix[m // len(matrix[0])][m % len(matrix[0])]
            if val == target:
                return True
            elif val < target:
                l = m+1
            else:
                r = m-1
        
        return False